File: templates/go/greeter.go
package main

import (
	restate "github.com/restatedev/sdk-go"
	"time"
)

// Greeter is a struct which represents a Restate service; reflection will turn exported methods into service handlers
type Greeter struct{}

func (Greeter) Greet(ctx restate.Context, name string) (string, error) {
    //  Durably execute a set of steps; resilient against failures
	greetingId := restate.Rand(ctx).UUID().String()

	if _, err := restate.Run(ctx, func(ctx restate.RunContext) (restate.Void, error) {
		return restate.Void{}, SendNotification(greetingId, name)
	}); err != nil {
		return "", err
	}

	if err := restate.Sleep(ctx, 1*time.Second); err != nil {
		return "", err
	}

	if _, err := restate.Run(ctx, func(ctx restate.RunContext) (restate.Void, error) {
		return restate.Void{}, SendReminder(greetingId)
	}); err != nil {
		return "", err
	}

    // Respond to caller
	return "You said hi to " + name + "!", nil
}

File: java/integrations/java-gradle-lambda-cdk/lambda/src/main/java/my/example/Greeter.java
package my.example;

import static my.example.Utils.sendNotification;
import static my.example.Utils.sendReminder;

import dev.restate.sdk.Restate;
import dev.restate.sdk.annotation.Handler;
import dev.restate.sdk.annotation.Service;
import java.time.Duration;

@Service
public class Greeter {

  @Handler
  public String greet(String name) {
    // Durably execute a set of steps; resilient against failures
    String greetingId = Restate.random().nextUUID().toString();
    Restate.run("send-notification", () -> sendNotification(greetingId, name));
    Restate.sleep(Duration.ofMillis(1000));
    Restate.run("send-reminder", () -> sendReminder(greetingId));

    // Respond to caller
    return "You said hi to " + name + "!";
  }
}
